pkg/tsdb/influxdb/response_parser.go

```diff
@@ -86,14 +88,25 @@ func transformRows(rows []Row, query *Query) data.Frames {
 			// set a nice name on the value-field
 			valueField.SetConfig(&data.FieldConfig{DisplayNameFromDS: name})
 
-			frames = append(frames, data.NewFrame(name, timeField, valueField))
+			frames = append(frames, newDataFrame(name, query.RawQuery, *timeField, *valueField))
 		}
 	}
 
 	return frames
 }
 
-func formatFrameName(row Row, column string, query *Query) string {
+func newDataFrame(name string, queryString string, timeField data.Field, valueField data.Field) *data.Frame {
+	frame := data.NewFrame(name, &timeField, &valueField)
+	if queryString != "" {
```

Comment:
do we have data showing that sometimes `queryString` is empty-string? in general i would just assume it is never empty, or it's not empty in 99% of cases, and i'd just always create the `FrameMeta`, but maybe there are often cases when it is not created?

Reply:
The `rawQuery` (`queryString`) is added to every query - so it can't be empty. Removed this empty-string check. 

https://github.com/grafana/grafana/pull/43352/files#diff-ed00ef1a9bc60de20f80690560c9ac30c8e1fd8533c8e5aaafc46ff96a32df8bR130